[Metricbeat autodiscover][Provider Kubernetes] Add condition to node/namespace watchers (#37181)
elastic/beats: CHANGELOG-developer.next.asciidoc

==== Bugfixes

> - Stop namespace and node watchers from starting on metricbeat autodiscover if `add_resource_metadata` is disabled.{pull}37181[37181]

Reviewer: ```suggestion
- Do not start namespace and node watchers on metricbeat autodiscover if `add_resource_metadata` is disabled.{pull}37181[37181]
```
